frontend/src/App.jsx

```diff
@@ -1,22 +1,48 @@
-import React from 'react'
+import { React, useState, useEffect } from 'react'
 import { Routes, Route } from 'react-router-dom'
+import axios from 'axios'
+import UserContext from './contexts/UserContext'
 import Layout from './components/Layout'
 import IndexPage from './pages/IndexPage'
 import LoginPage from './pages/LoginPage'
 import SignupPage from './pages/SignupPage'
 
 
 const App = () => {
+  const [user, setUser] = useState('')
+
+  useEffect(() => {
+    const checkAuth = async () => {
+      const res = await axios.get('http://localhost:2003/user', {
```

Comment:
Clarifying my comment about environment variables from above (since it applies here too):

If you have the domain you're using (localhost, sunscreensavvy.com or anything else) as an environment variable, then you can swap them out depending on the environment you have your code deployed to-- perhaps on your local machine it's just localhost, and the production version of the site is on the domain-named url.

So basically instead of "get('localhost:2003/user')" it'd be something like get("{BACKEND_ROUTE}/user" where BACKEND_ROUTE evaluates to your url.

 (This might not be too useful yet, but might be good to keep in mind for the future)